apps/api/migrations/093_add_ledger_sequence_to_derived_events.up.sql

```
@@ -0,0 +1,19 @@
+-- long after the fact would insert rows with a recent occurred_at for old
+-- ledgers, making "clear rows for this ledger range" impossible to express
+-- correctly. ledger_sequence is populated by the indexer's apply* handlers
+-- going forward (see internal/stellar/indexer.go) and is what
+-- Runner.resetScope now filters DELETEs on, instead of clearing every row
+-- for a contract regardless of range.
+ALTER TABLE penalty_events ADD COLUMN IF NOT EXISTS ledger_sequence BIGINT;
+ALTER TABLE penalty_distributions ADD COLUMN IF NOT EXISTS ledger_sequence BIGINT;
+ALTER TABLE vault_rebalance_legs ADD COLUMN IF NOT EXISTS ledger_sequence BIGINT;
+ALTER TABLE vault_rebalance_completions ADD COLUMN IF NOT EXISTS ledger_sequence BIGINT;
+
+CREATE INDEX IF NOT EXISTS idx_penalty_events_ledger_sequence ON penalty_events (ledger_sequence);
```

SQLFluff (4.2.2)

[error] CREATE INDEX should use CONCURRENTLY to avoid locking the table during the build.

(PG01)